internal/broadcaster/rate_broadcaster.go

```diff
@@ -158,18 +160,67 @@ utxoLoop:
 			return txs, nil
 		case utxo := <-b.utxoCh:
 			tx := sdkTx.NewTransaction()
+			amount := utxo.Satoshis
 
 			err := tx.AddInputsFromUTXOs(utxo)
 			if err != nil {
 				return nil, errors.Join(ErrFailedToAddInput, err)
 			}
 
-			fee, err := b.feeModel.ComputeFee(tx)
+			if b.opReturn != "" {
+				err = tx.AddOpReturnOutput([]byte(b.opReturn))
+				if err != nil {
+					return nil, fmt.Errorf("failed to add OP_RETURN output: %v", err)
+				}
+			}
+
+			if b.maxSize > 0 {
+				// Add additional inputs to the transaction
+				randomInt, err := rand.Int(rand.Reader, big.NewInt(int64(10)))
```

Comment:
This `10` here is a magic number. `randomInt` is also not very informative. Maybe `numOfInputs`?

Reply:
Done